File: client/python-module/secret_client.py
import requests

class SecretClient:
    def __init__(self, base_url: str, token: str = None):
        self.base_url = base_url.rstrip("/")
        self.headers = {"Authorization": f"Bearer {token}"} if token else {}

    def save_raw(self, namespace: str, key: str, value: str):
        url = f"{self.base_url}/api/secrets/raw"
        payload = {"namespace": namespace, "key": key, "value": value}
        resp = requests.post(url, json=payload, headers=self.headers)
        resp.raise_for_status()
        return resp.json()

    def save_encrypted(self, namespace: str, key: str, value: str):
        url = f"{self.base_url}/api/secrets/encrypted"
        payload = {"namespace": namespace, "key": key, "value": value}
        resp = requests.post(url, json=payload, headers=self.headers)
        resp.raise_for_status()
        return resp.json()

    def get_raw(self, namespace: str, key: str):
        url = f"{self.base_url}/api/secrets/raw/{namespace}/{key}"
        resp = requests.get(url, headers=self.headers)
        resp.raise_for_status()
        return resp.json()

    def get_decrypted(self, namespace: str, key: str):
        url = f"{self.base_url}/api/secrets/decrypted/{namespace}/{key}"
        resp = requests.get(url, headers=self.headers)
        resp.raise_for_status()
        return resp.json()

File: client/nodejs-module/secretClient.js
// secretClient.js
const axios = require('axios');

class SecretClient {
    constructor(baseURL, token) {
        this.baseURL = baseURL;
        this.headers = token ? { Authorization: `Bearer ${token}` } : {};
    }

    async saveRaw(namespace, key, value) {
        const res = await axios.post(`${this.baseURL}/api/secrets/raw`, { namespace, key, value }, { headers: this.headers });
        return res.data;
    }

    async saveEncrypted(namespace, key, value) {
        const res = await axios.post(`${this.baseURL}/api/secrets/encrypted`, { namespace, key, value }, { headers: this.headers });
        return res.data;
    }

    async getRaw(namespace, key) {
        const res = await axios.get(`${this.baseURL}/api/secrets/raw/${namespace}/${key}`, { headers: this.headers });
        return res.data;
    }

    async getDecrypted(namespace, key) {
        const res = await axios.get(`${this.baseURL}/api/secrets/decrypted/${namespace}/${key}`, { headers: this.headers });
        return res.data;
    }
}

module.exports = SecretClient;